test/test_bdecode.cpp

```diff
@@ -848,9 +849,12 @@ TORRENT_TEST(dict_find_funs)
 	TEST_CHECK(!e.dict_find_dict("c"));
 
 	// variants taking std::string
-	TEST_EQUAL(e.dict_find_dict(std::string("d")).dict_find_int_value("x"), 1);
-	TEST_CHECK(!e.dict_find_dict(std::string("c")));
-	TEST_CHECK(!e.dict_find_dict(std::string("x")));
+	std::string d_str("d");
+	std::string c_str("c");
+	std::string x_str("x");
+	TEST_EQUAL(e.dict_find_dict(d_str).dict_find_int_value("x"), 1);
```

Comment:
this test block was made especially for std::string api call  "// variants taking std::string"